vllm/engine/metrics.py

```diff
@@ -669,20 +670,37 @@ def log(self, stats: Stats):
         if local_interval_elapsed(stats.now, self.last_local_log,
                                   self.local_interval):
             if self.spec_decode_metrics is not None:
+                # The counters in self.spec_decode_metrics are aggregates.
+                # The Prometheus Counters must be incremented with deltas.
+                # Keep track of the previous value so we can compute deltas.
```

Comment:
I'd be inclined to do something like this:

```
def log_counter_from_cumulative(metric, cumulative, previous):
    self._log_counter(metric, cumulative - previous)
```

The method name helps the code to be self-documenting, requiring less comments

and then:

```
    self._log_from_cumulative(
        self.metrics.counter_spec_decode_num_accepted_tokens,
        self.spec_decode_metrics.accepted_tokens,
	self.spec_decode_prev_num_accepted)
    self.spec_decode_prev_num_accepted = self.spec_decode_metrics.accepted_tokens
```

since we don't need all of SpecDecodeWorkerMetrics